builtin/providers/aws/resource_aws_db_option_group.go

```diff
@@ -0,0 +1,295 @@
+			},
+			"option_group_description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"option": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"option_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"db_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"vpc_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+				Set: resourceAwsDbOptionHash,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDbOptionGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	createOpts := &rds.CreateOptionGroupInput{
+		EngineName:             aws.String(d.Get("engine_name").(string)),
+		MajorEngineVersion:     aws.String(d.Get("major_engine_version").(string)),
+		OptionGroupDescription: aws.String(d.Get("option_group_description").(string)),
+		OptionGroupName:        aws.String(d.Get("name").(string)),
+		Tags:                   tags,
+	}
+
+	log.Printf("[DEBUG] Create DB Option Group: %#v", createOpts)
+	_, err := rdsconn.CreateOptionGroup(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DB Option Group: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	log.Printf("[INFO] DB Option Group ID: %s", d.Id())
+
+	return resourceAwsDbOptionGroupUpdate(d, meta)
+}
+
+func resourceAwsDbOptionGroupRead(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	params := &rds.DescribeOptionGroupsInput{
+		OptionGroupName: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Describe DB Option Group: %#v", params)
+	options, err := rdsconn.DescribeOptionGroups(params)
+	if err != nil {
+		return fmt.Errorf("Error Describing DB Option Group: %s", err)
+	}
+
+	var option *rds.OptionGroup
+	for _, ogl := range options.OptionGroupsList {
+		if *ogl.OptionGroupName == d.Get("name").(string) {
+			option = ogl
+			break
+		}
+	}
+
+	if option == nil {
+		return fmt.Errorf("Unable to find Option Group: %#v", options.OptionGroupsList)
+	}
+
+	d.Set("major_engine_version", options.OptionGroupsList[0].MajorEngineVersion)
+	d.Set("engine_name", options.OptionGroupsList[0].EngineName)
+	d.Set("option_group_description", options.OptionGroupsList[0].OptionGroupDescription)
```

Comment:
On L128 we pull the option out of the list into a local var `option` - might as well use it here rather than just reaching back into the `OptionGroupList[]`.